charts/metallb/templates/controller.yaml

```diff
@@ -5,12 +5,12 @@ metadata:
   name: {{ template "metallb.fullname" . }}-controller
   labels:
     {{- include "metallb.labels" . | nindent 4 }}
-    app.kubernetes.io/component: controller
+    component: controller
```

Comment:
What's the reason for changing this?  `app.kubernetes.io/component` comes from the recommended labels: https://kubernetes.io/docs/concepts/overview/working-with-objects/common-labels/

Reply:
If the E2E tests are looking for `component`, they should probably be changed to reflect the standard expected by the Kubernetes documentation.

Reply:
Agree. I _think_ e2e tests are crafted against the deployment via the manifest, which if I remember correctly use the `component: controller` .
It's probably worth fixing the manifests.

Reply:
Does it make sense to transition to using the helm chart to generate the manifests, via `helm template`.  I believe projects like calico do this, as well.

Reply:
Yep, let me file an issue for that. It would be nice to have CI that ensure they do not drift.

Reply:
We can provide different manifest flavors by templating with different `values.yaml` files.  The `charts/metallb/values.yaml` file is the one that gets officially "packaged" in the chart, but we can put others in the same directory as reference/manifest sources.

Reply:
Neat, I hoped that could have been done as we are gonna need at least a couple of flavours (with / without FRR).
Thanks @gclawes !

Reply:
ok i will update it the label in the manifests and E2E tests.